src/ocsp.c

```diff
@@ -928,6 +946,87 @@ int wolfSSL_i2d_OCSP_REQUEST_bio(WOLFSSL_BIO* out,
     return WOLFSSL_FAILURE;
 }
 #endif /* !NO_BIO */
+
+int wolfSSL_i2d_OCSP_CERTID(WOLFSSL_OCSP_CERTID* id, unsigned char** data)
+{
+    if (id == NULL || data == NULL)
+        return WOLFSSL_FAILURE;
+
+    if (*data != NULL) {
+        XMEMCPY(*data, id->rawCertId, id->rawCertIdSize);
+        *data = *data + id->rawCertIdSize;
+    }
+    else {
+        *data = XMALLOC(id->rawCertIdSize, NULL, DYNAMIC_TYPE_OPENSSL);
+        XMEMCPY(*data, id->rawCertId, id->rawCertIdSize);
+    }
+
+    return id->rawCertIdSize;
+}
+
+const WOLFSSL_OCSP_CERTID* wolfSSL_OCSP_SINGLERESP_get0_id(const WOLFSSL_OCSP_SINGLERESP *single)
+{
+    return single;
```

Comment:
WOLFSSL_OCSP_CERTID and WOLFSSL_OCSP_SINGLERESP are represented by the same OcspEntry. Hence, just return the argument.

Reply:
You'll need a cast here.

Reply:
Do I need it if WOLFSSL_OCSP_CERTID and WOLFSSL_OCSP_SINGLERESP are the same type? They're both typedefs for OcspEntry.